Unraid Docker Crash

2019/10/01 22:50:22 [INFO] UniFi Measurements Recorded. Sites: 129, Clients: 1673, Wireless APs: 427, Gateways: 0, Switches: 38, Points: 5145, Fields: 220947
2019/10/01 22:51:08 [ERROR] (1/0) unifi.GetDevices(): Get https://unifi.go2rcg.com:8443/api/s/jy2o2a8o/stat/device: dial tcp: lookup unifi.go2rcg.com on 1.0.0.1:53: read udp 172.17.0.14:54539->1.0.0.1:53: i/o timeout
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x720ad8]

goroutine 1 [running]:
github.com/davidnewhall/unifi-poller/unifipoller.(*UnifiPoller).AugmentMetrics(0xc0000d2280, 0xc000116000, 0x0, 0x0)
/go/src/github.com/davidnewhall/unifi-poller/unifipoller/unifi.go:121 +0x108
github.com/davidnewhall/unifi-poller/unifipoller.(*UnifiPoller).CollectAndReport(0xc0000d2280, 0xc001ab1d50, 0x1)
/go/src/github.com/davidnewhall/unifi-poller/unifipoller/unifi.go:83 +0x5a
github.com/davidnewhall/unifi-poller/unifipoller.(*UnifiPoller).PollController(0xc0000d2280, 0x6, 0xc000016050)
/go/src/github.com/davidnewhall/unifi-poller/unifipoller/unifi.go:63 +0x25e
github.com/davidnewhall/unifi-poller/unifipoller.(*UnifiPoller).Run(0xc0000d2280, 0x0, 0x0)
/go/src/github.com/davidnewhall/unifi-poller/unifipoller/start.go:93 +0x4ec
github.com/davidnewhall/unifi-poller/unifipoller.Start(0x40576f, 0xc0000b8058)
/go/src/github.com/davidnewhall/unifi-poller/unifipoller/start.go:51 +0x31d
main.main()
/go/src/github.com/davidnewhall/unifi-poller/main.go:11 +0x26

--- a/unifipoller/unifi.go
+++ b/unifipoller/unifi.go
@@ -62,9 +62,8 @@ func (u *UnifiPoller) PollController() error {
 			// Only run this if the authentication procedure didn't return error.
 			_ = u.CollectAndReport()
 		}
-		if u.Config.MaxErrors >= 0 && u.errorCount > u.Config.MaxErrors {
-			return fmt.Errorf("reached maximum error count, stopping poller (%d > %d)",
-				u.errorCount, u.Config.MaxErrors)
+		if u.errorCount > 0 {
+			return fmt.Errorf("controller or influxdb errors, stopping poller")
 		}
 	}
 	return nil
@@ -116,6 +115,9 @@ func (u *UnifiPoller) CollectMetrics() (*Metrics, error) {
 // This is where we can manipuate the returned data or make arbitrary decisions.
 // This function currently adds parent device names to client metrics.
 func (u *UnifiPoller) AugmentMetrics(metrics *Metrics) error {
+	if metrics == nil || metrics.Devices == nil || metrics.Clients == nil {
+		return fmt.Errorf("nil metrics, augment impossible")
+	}
 	devices := make(map[string]string)
 	bssdIDs := make(map[string]string)
 	for _, r := range metrics.UAPs {

